src/Compilers/CSharp/Portable/Errors/LazyObsoleteDiagnosticInfo.cs

```diff
@@ -29,7 +29,7 @@ internal override DiagnosticInfo GetResolvedInfo()
             {
                 // A symbol's Obsoleteness may not have been calculated yet if the symbol is coming
                 // from a different compilation's source. In that case, force completion of attributes.
-                var symbol = (_symbolOrSymbolWithAnnotations as Symbol) ?? ((SymbolWithAnnotations)_symbolOrSymbolWithAnnotations).Symbol;
+                var symbol = (_symbolOrSymbolWithAnnotations as Symbol) ?? ((TypeSymbolWithAnnotations)_symbolOrSymbolWithAnnotations).TypeSymbol;
```

Comment:
>TypeSymbolWithAnnotations [](start = 77, length = 25)

How do we know this assert will succeed?
Consider adding an assertion in the constructor.